packages/react-native-otlp/test-project/ios/RNEmbraceOTLPTests/RNEmbraceOTLPTests.swift

```diff
@@ -0,0 +1,209 @@
+        rejectCalls.append(b!)
+    }
+
+    func reset() {
+        resolveCalls.removeAll()
+        rejectCalls.removeAll()
+    }
+}
+
+let SDK_BASE_CONFIG: NSDictionary = [
+    "appId": "abcde",
+    "endpointBaseUrl": "http://localhost/dev/null"
+]
+
+final class RNEmbraceOTLPTests: XCTestCase {
+    var module: RNEmbraceOTLP!
+    var promise: Promise!
+
+    override func setUp() async throws {
+        module = RNEmbraceOTLP()
+        promise = Promise()
+    }
+
+    // happy path
+    func testStartNativeEmbraceSDK() throws {
+        let expectation = self.expectation(description: "testStartNativeEmbraceSDK")
+
+        let traceExportConfig: NSDictionary = [
+            "endpoint": "https://test-trace-endpoint/v1",
+            "headers": [("Authorization", "base64_instance:token")],
+            "timeout": NSNumber(200000)
+        ]
+
+        let logExportConfig: NSDictionary = [
+            "endpoint": "https://test-log-endpoint/v1",
+            "headers": [("Authorization", "base64_instance:token")],
+            "timeout": NSNumber(300000)
+        ]
+
+        module.startNativeEmbraceSDK(sdkConfigDict: SDK_BASE_CONFIG,
+                                     otlpExportConfigDict: [
+                                        "traceExporter": traceExportConfig,
+                                        "logExporter": logExportConfig
+                                    ],
+                                     resolve: { result in
+                                        self.promise.resolve(val: result)
+                                        expectation.fulfill()
+                                    },
+                                     rejecter: promise.reject)
+
+        waitForExpectations(timeout: 3)
+        XCTAssertEqual(promise.resolveCalls.count, 1)
+    }
+
+    func testStartWithMissingExporters() throws {
+        let expectation = self.expectation(description: "testStartWithMissingExporters")
+        
+        module.startNativeEmbraceSDK(sdkConfigDict: SDK_BASE_CONFIG,
+                                     otlpExportConfigDict: [:], // empty config
+                                     resolve: { result in
+                                        self.promise.resolve(val: result)
+                                        expectation.fulfill()
+                                    },
+                                     rejecter: promise.reject)
+
+        waitForExpectations(timeout: 3)
+        
+        // TODO: Test that log is printed
+        // [Embrace] Neither Traces nor Logs configuration were found, skipping custom export.
+        XCTAssertEqual(promise.resolveCalls.count, 1)
+    }
+
+    func testStartWithTraceExporterOnly() throws {
+        let expectation = self.expectation(description: "testStartWithTraceExporterOnly")
+        expectation.expectedFulfillmentCount = 4
+        
+        // happy path only with Trace config
+        module.startNativeEmbraceSDK(sdkConfigDict: SDK_BASE_CONFIG,
```

Comment:
may need to rethink how these tests are structured, in the tracer provider + core packages I was seeing flaky behaviour in the tests runs whenever the Embrace SDK was started multiple times. This was due to the fact that there's not a way to stop the SDK so previous instances of it could interfere with subsequent cases

some options:
* have only a single successful case where the SDK is started, do the failing cases prior to that
* have the tests work on a helper that has most of the setup logic but doesn't actually start the SDK

though actually now that I check it seems like the IOS tests have been passing fine so maybe it is not an issue with these particular cases

Reply:
yeah, so far I did not see any failure. maybe I can revisit it if we start to see issues there